tmpl/wiki/deployment.md

```diff
@@ -146,64 +161,90 @@ a small `travis-senv` command-line tool to make this easier.
+  ?|~ "echo '%s' > ~/.ssh/config" ssh_config;
+  (* configure git for github *)
+  ?|  "git config --global user.email 'travis@openmirage.org'";
+  ?|  "git config --global user.name 'Travis the Build Bot'";
+  ?|  "git config --global push.default simple";
+  (* clone deployment repo *)
+  ?|  "git clone git@mir-deploy:${TRAVIS_REPO_SLUG}-deployment";
+  (* remove and recreate any existing image for this commit *)
+  ?|  "rm -rf $DEPLOYD/xen/$TRAVIS_COMMIT";
+  ?|  "mkdir -p $DEPLOYD/xen/$TRAVIS_COMMIT";
+  ?|  "cp $MIRDIR/$XENIMG $MIRDIR/config.ml $DEPLOYD/xen/$TRAVIS_COMMIT";
+  ?|  "bzip2 -9 $DEPLOYD/xen/$TRAVIS_COMMIT/$XENIMG";
+  ?|  "echo $TRAVIS_COMMIT > $DEPLOYD/xen/latest";
+  (* commit and push changes *)
+  ?|  "cd $DEPLOYD &&\
+       \ git add xen/$TRAVIS_COMMIT xen/latest &&\
+       \ git commit -m \"adding $TRAVIS_COMMIT for $MIRAGE_BACKEND\" &&\
+       \ git push"
+end
 ```
 
-While this fragment is a bit longer than the rest, it's all rather
-straightforward.  We install `travis-senv` and use it to retrieve
-our SSH deployment key (which is stored encrypted in the `.travis.yml`
-file, and can only be decrypted by the Travis infrastructure).
-Once that's installed, we initialize `git` and clone the deployment
-repository.  We then compress our output kernel and commit it to
-the repository along with the configuration data.
+(The `?|` and `?|~` operators are defined in [`yorick.ml`](https://github.com/ocaml/ocaml-travisci-skeleton/blob/master/yorick.ml) and act simply as shell
+invocations piping `stdin`/`stdout`/`stderr` in the usual way, with
+success/failure tests on the status of each invoked command.)
+
+While this fragment is a bit longer than the rest, it's actually rather
+straightforward. We install `travis-senv` and use it to retrieve our SSH
+deployment key (which is stored encrypted in the `.travis.yml` file, and can
+only be decrypted by the Travis infrastructure). Once that's installed, we
+initialise `git` and clone the deployment repository. We then compress our
+output kernel, update the `xen/latest` pointer to refer to the new kernel,
+commit everything, and push the results back.
 
 ### Updating the live website
 
-The live website runs on a Xen machine hosted in Cambridge, and
-this simply polls the [mirage/mirage-www-deployment](http://github.com/mirage/mirage-www-deployment)
-repository for new pushes, and updates to the latest version in
-that repository when one appears.  This machine doesn't need to
-have any of the MirageOS tools installed, and if there's an error in
-an updated website, we can simply revert back to a previous changeset
-of the deployed site.
+The live website runs on a Xen machine which has a crontab entry in dom0 that
+polls the
+[mirage/mirage-www-deployment](http://github.com/mirage/mirage-www-deployment)
+repository for new content via `git pull`. When new content __is__ obtained, a
+Git [`merge-hook`](https://github.com/mirage/mirage-www-deployment) is invoked
```

Comment:
`merge-hook` -> `post-merge` hook

Maybe explicitly reference the scripts here and link to them in the deploy repo? There may be folks who aren't as familiar with git's hooks so I wonder if linking to the manual would be useful (http://git-scm.com/docs/githooks#_post_merge)